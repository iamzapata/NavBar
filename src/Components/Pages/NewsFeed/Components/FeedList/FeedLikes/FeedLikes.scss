@import "styles/variables.scss";

.FeedLikes {
  .icon {
    border: solid 1px $color-transparent-gray-border;
    box-shadow: 0px 0px 1px $color-gray-border;
  }
  .Likes__Count span {
    color: $color-light-gray-text;
    margin: 0 0 0 10px;
  }
}
