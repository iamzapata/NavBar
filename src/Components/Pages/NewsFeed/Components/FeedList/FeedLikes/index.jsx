import React from "react"
import { shape, number } from "prop-types"
import "./FeedLikes.scss"

const colleagues = likes => (likes > 1 ? "Colleagues" : "Colleague")

const FeedLikes = ({ feed: { likes } }) => (
  <span className="FeedLikes">
    <span className="Likes__Icon">
      <span className="icon">
        {likes > 0 && <i className="fa fa-heart" />}
        {likes === 0 && <i className="fa fa-heart-o" />}
      </span>
    </span>
    <span className="Likes__Count is-size-6">
      {likes > 0 && (
        <span>
          {`${likes} `}
          {colleagues(likes)}
          {" liked this"}
        </span>
      )}
    </span>
  </span>
)

FeedLikes.propTypes = {
  feed: shape({
    likes: number
  }).isRequired
}

export default FeedLikes
