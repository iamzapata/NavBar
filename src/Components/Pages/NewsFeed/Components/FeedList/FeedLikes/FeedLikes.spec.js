import React from "react"
import { render, cleanup } from "react-testing-library"
import FeedLikes from "./index"

describe("<FeedLikes />", () => {
  afterEach(cleanup)
  it("Should render self", () => {
    const { container } = render(<FeedLikes />)
    expect(container.firstChild).toBeTruthy()
    expect(container.firstChild).toMatchSnapshot()
  })
})
