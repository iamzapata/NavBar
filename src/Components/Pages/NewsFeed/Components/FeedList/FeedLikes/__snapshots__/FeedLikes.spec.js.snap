// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FeedLikes /> Should render self 1`] = `
<span
  class="FeedLikes"
>
  <span
    class="Likes__Icon"
  >
    <span
      class="icon"
    />
  </span>
  <span
    class="Likes__Count is-size-6"
  />
</span>
`;
